[Bug] 빌드 오류 Error: Command "pnpm run build" exited with 1

## 🐛 버그 설명
src/common/hooks/useBottomSheetStore.ts(1,10): error TS1484: 'ReactNode' is a type and must be imported using a type-only import when 'verbatimModuleSyntax' is enabled.
src/features/shelter/index.ts(2,15): error TS2305: Module '"./components/ShelterBottomSheetContent"' has no exported member 'ShelterItem'.
vite.config.ts(6,29): error TS2580: Cannot find name 'process'. Do you need to install type definitions for node? Try `npm i --save-dev @types/node`.
 ELIFECYCLE  Command failed with exit code 1.
Error: Command "pnpm run build" exited with 1

## 🔍 재현 단계
1. vercel 배포 과정에서 발생

## 🎯 예상 결과
정상 배포

--- a/src/common/components/BottomSheetView.tsx
+++ b/src/common/components/BottomSheetView.tsx
@@ -22,7 +22,6 @@ export function BottomSheetView(props: BottomSheetViewProps) {
   const {
     translateY,
     isDragging,
-    isAnimating,
     measured,
     sheetRef,
     handleRef,

--- a/src/common/hooks/useBottomSheetStore.ts
+++ b/src/common/hooks/useBottomSheetStore.ts
@@ -1,4 +1,4 @@
-import { ReactNode } from "react";
+import type { ReactNode } from "react";
 import { create } from "zustand";
 
 export type BottomSheetOptions = {

--- a/src/features/map/hooks/useCurrentPosition.ts
+++ b/src/features/map/hooks/useCurrentPosition.ts
@@ -1,10 +1,5 @@
 import { useCurrentPositionQuery, getPositionError } from "./useCurrentPositionQuery";
 
-type PositionError = {
-  type: "permission_denied" | "position_unavailable" | "timeout" | "not_supported" | "unknown";
-  message: string;
-};
-
 export function useCurrentPosition() {
   const { data, error, isLoading, refetch } = useCurrentPositionQuery();
 

--- a/src/features/map/hooks/useMapBottomSheet.tsx
+++ b/src/features/map/hooks/useMapBottomSheet.tsx
@@ -1,10 +1,7 @@
-import React, { useEffect, useRef } from "react";
+import { useEffect, useRef } from "react";
 import { useBottomSheetStore } from "@/common/hooks/useBottomSheetStore";
 import { ShelterBottomSheetContent } from "@/features/shelter";
-import type {
-  NearbyShelterApiItem,
-  NearbyShelterApiResponse
-} from "@/features/shelter/schemas/shelter.schema";
+import type { NearbyShelterApiItem } from "@/features/shelter/schemas/shelter.schema";
 
 /**
  * 지도에서 바텀시트를 관리하는 로직을 제공하는 훅

--- a/src/features/map/services/geolocation.ts
+++ b/src/features/map/services/geolocation.ts
@@ -4,8 +4,8 @@ export function getCurrentPositionOnce(options?: Options) {
   return new Promise<GeolocationPosition>((resolve, reject) => {
     if (!navigator.geolocation) {
       const error = new GeolocationPositionError();
-      error.code = 0; // UNKNOWN_ERROR
-      error.message = "Geolocation not supported";
+      Object.defineProperty(error, "code", { value: 0 }); // UNKNOWN_ERROR
+      Object.defineProperty(error, "message", { value: "Geolocation not supported" });
       reject(error);
       return;
     }

--- a/src/features/shelter/components/ShelterBottomSheetContent.tsx
+++ b/src/features/shelter/components/ShelterBottomSheetContent.tsx
@@ -3,6 +3,8 @@ import { useNavigate } from "react-router-dom";
 import type { NearbyShelterApiItem } from "../schemas/shelter.schema";
 import { useBottomSheetStore } from "@/common/hooks/useBottomSheetStore";
 
+export type ShelterItem = NearbyShelterApiItem;
+
 export function ShelterBottomSheetContent({
   items,
   error

